### PR TITLE
Fixed issue where using Update-AzureRmSqlDatabaseVulnerabilityAssessmentSettings with storage account name parameter set

### PR DESCRIPTION
**Pull request by yaakoviyun**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by yaakoviyun** [comment]
@yoavfr @talhers

File: src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.Netcore.csproj [hunk 1]
```diff
@@ -24,7 +24,7 @@
   </PropertyGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' != 'net472'">
-    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.18.1-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.21.0-preview" />
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.Netcore.csproj]: You will need to re-publish this client for the new runtime and newtonsoft.json version

**Comment by jaredmoo** [comment]
https://github.com/Azure/azure-sdk-for-net/pull/4984

**Comment by jaredmoo** [comment]
https://www.nuget.org/packages/Microsoft.Azure.Management.Sql/1.22.0-preview

**Comment by yaakoviyun** [comment]
@markcowl - done

**Comment by yaakoviyun** [comment]
@markcowl  - Hi, 
can you merge?

File: src/ResourceManager/Sql/Commands.Sql/ChangeLog.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Current Release
+* Fixed issue where using Update-AzureRmSqlDatabaseVulnerabilityAssessmentSettings with storage account name parameter set
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/ChangeLog.md]: This message isn't clear - it seems like you either didn't finish the sentence or meant "Fixed issue with using Update-AzureRmSqlDatabaseVulnerabilityAssessmentSettings with storage account name parameter set".  Please fix.

File: src/ResourceManager/Sql/Commands.Sql/VulnerabilityAssessment/Services/SqlVulnerabilityAssessmentAdapter.cs [hunk 3]
```diff
@@ -83,5 +73,33 @@ public StorageContainerInfo CreateBlobStorageContainerAndSAS(string storageAccou
             return ConvertSettingsToModel(model.ResourceGroupName, model.ServerName, model.DatabaseName, vulnerabilityAssessmentSettings);
 
         }
+		
+	/// <summary>
+        /// Set settings with storage access key
+        /// </summary>
+        public DatabaseVulnerabilityAssessmentSettingsModel SetDatabaseVulnerabilityAssessmentSettingsWithStorageAccesKey(DatabaseVulnerabilityAssessmentSettingsModel model)
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/VulnerabilityAssessment/Services/SqlVulnerabilityAssessmentAdapter.cs]: "AccessKey" - missing an "s"

**Comment by yaakoviyun** [comment]
@maddieclayton - done.

**Comment by yaakoviyun** [comment]
Will fix the tests i a different PR

**Comment by yaakoviyun** [comment]
@markcowl  and @maddieclayton  - I've fixed all the issues.
Can you review and merge if everything looks OK?

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/VulnerabilityAssessmentTests.ps1 [hunk 4]
```diff
@@ -179,34 +163,17 @@ function Test-VulnerabilityAssessmentSettingsTest
 		Assert-AreEqual "vulnerability-assessment" $settings.ScanResultsContainerName
 		Assert-AreEqual $params.storageAccount $settings.StorageAccountName	
 		Assert-AreEqual None $settings.RecurringScansInterval
-		Assert-AreEqual $false $settings.EmailAdmins
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/VulnerabilityAssessmentTests.ps1]: Why did this get changed?

File: src/ResourceManager/Sql/Commands.Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/UpdateAzureSqlDatabaseVulnerabilityAssessmentSettings.cs [hunk 5]
```diff
@@ -226,28 +226,19 @@ protected override DatabaseVulnerabilityAssessmentSettingsModel ApplyUserInputTo
...
             }
             else 
             {
-                // SetSettingsWithStorageAccountNameParameterSet and SetByInputObjectParameterSet
-                StorageContainerInfo storageContainerInfo = ModelAdapter.CreateBlobStorageContainerAndSAS(model.StorageAccountName,
-                    model.ScanResultsContainerName);
-
-                storageContainerPath = storageContainerInfo.StorageContainerPath;
-                storageAccountSasKey = storageContainerInfo.StorageAccountSasKey;
+				return ModelAdapter.SetDatabaseVulnerabilityAssessmentSettingsWithStorageAccessKey(model);
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/UpdateAzureSqlDatabaseVulnerabilityAssessmentSettings.cs]: How does the behavior change here? Is it exactly the same for users? It seems like the storage container is no longer being created - does it have to already exist?

**Comment by yaakoviyun** [comment]
First change was to fix a test record that was wrongly recorded in addition to the backend change for default settings.

For the second comment, our backend now creates the container, so there is no change for the user experience l.

Sent from Outlook
________________________________
מאת: Maddie Clayton <notifications@github.com>
‏‏נשלח: יום שלישי 06 נובמבר 2018 00:12:07
‏‏אל: Azure/azure-powershell
עותק: Yaakov Iyun; Mention
‏‏נושא: Re: [Azure/azure-powershell] Fixed issue where using Update-AzureRmSqlDatabaseVulnerabilityAssessmentSettings with storage account name parameter set (#7715)


@maddieclayton commented on this pull request.

________________________________

In src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/VulnerabilityAssessmentTests.ps1<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-powershell%2Fpull%2F7715%23discussion_r230930937&data=02%7C01%7Cyaiyun%40microsoft.com%7C15b9db4571464663505a08d6436bba0f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636770527304576425&sdata=AxiUZnZ5Z6WQp%2FhK9Rn8jWhnJyWrVc59h%2FcrWBT5sK8%3D&reserved=0>:

> @@ -179,34 +163,17 @@ function Test-VulnerabilityAssessmentSettingsTest
                Assert-AreEqual "vulnerability-assessment" $settings.ScanResultsContainerName
                Assert-AreEqual $params.storageAccount $settings.StorageAccountName
                Assert-AreEqual None $settings.RecurringScansInterval
-               Assert-AreEqual $false $settings.EmailAdmins


Why did this get changed?

________________________________

In src/ResourceManager/Sql/Commands.Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/UpdateAzureSqlDatabaseVulnerabilityAssessmentSettings.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-powershell%2Fpull%2F7715%23discussion_r230932357&data=02%7C01%7Cyaiyun%40microsoft.com%7C15b9db4571464663505a08d6436bba0f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636770527304576425&sdata=c57rZErsdV3CIAWkKPi2yxovJuqDKkzUxAyYtFN2xsQ%3D&reserved=0>:

>              }
             else
             {
-                // SetSettingsWithStorageAccountNameParameterSet and SetByInputObjectParameterSet
-                StorageContainerInfo storageContainerInfo = ModelAdapter.CreateBlobStorageContainerAndSAS(model.StorageAccountName,
-                    model.ScanResultsContainerName);
-
-                storageContainerPath = storageContainerInfo.StorageContainerPath;
-                storageAccountSasKey = storageContainerInfo.StorageAccountSasKey;
+                               return ModelAdapter.SetDatabaseVulnerabilityAssessmentSettingsWithStorageAccessKey(model);


How does the behavior change here? Is it exactly the same for users? It seems like the storage container is no longer being created - does it have to already exist?

**Pull request merged**